service/chartconfig/v1/helm/helm_test.go

```diff
@@ -7,17 +7,73 @@ import (
 	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
 	"github.com/giantswarm/micrologger/microloggertest"
 	helmclient "k8s.io/helm/pkg/helm"
+	helmrelease "k8s.io/helm/pkg/proto/hapi/release"
 )
 
 func Test_GetReleaseContent(t *testing.T) {
 	testCases := []struct {
 		description     string
 		obj             v1alpha1.ChartConfig
+		releases        []*helmrelease.Release
 		expectedRelease *Release
 		errorMatcher    func(error) bool
 	}{
 		{
-			description: "case 0: chart not found",
+			description: "case 0: basic match with deployed status",
+			obj: v1alpha1.ChartConfig{
+				Spec: v1alpha1.ChartConfigSpec{
+					Chart: v1alpha1.ChartConfigSpecChart{
+						Name:    "quay.io/giantswarm/chart-operator-chart",
+						Channel: "0.1-beta",
+						Release: "chart-operator",
+					},
+				},
+			},
+			releases: []*helmrelease.Release{
+				helmclient.ReleaseMock(&helmclient.MockReleaseOptions{
+					Name:      "chart-operator",
+					Namespace: "default",
+				}),
+			},
+			expectedRelease: &Release{
+				Name:   "chart-operator",
+				Status: "DEPLOYED",
+				Values: map[string]interface{}{
+					// Note: Values cannot be configured via the Helm mock client.
+					"name": "value",
+				},
+			},
+			errorMatcher: nil,
+		},
+		{
+			description: "case 1: basic match with failed status",
+			obj: v1alpha1.ChartConfig{
+				Spec: v1alpha1.ChartConfigSpec{
+					Chart: v1alpha1.ChartConfigSpecChart{
+						Name:    "quay.io/giantswarm/chart-operator-chart",
+						Channel: "0.1-beta",
+						Release: "chart-operator",
+					},
+				},
+			},
+			releases: []*helmrelease.Release{
+				helmclient.ReleaseMock(&helmclient.MockReleaseOptions{
+					Name:       "chart-operator",
+					Namespace:  "default",
+					StatusCode: helmrelease.Status_FAILED,
```

Comment:
The Helm client has an enumeration for the status which I like. Not certain yet but I think our Helm client package could also expose this to the `chart` resource.